src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,74 @@
+import classNames from 'classnames';
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  left: number;
+  completed: number;
+  onFilter: (type: Filter) => void;
+  onClearCompleted: ()=>void;
+  currentFilter: Filter;
+};
+
+export const Footer: React.FC<Props> = ({
+  left, onFilter, completed, onClearCompleted, currentFilter,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${left} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={classNames(
+            'filter__link', { selected: currentFilter === Filter.ALL },
+          )}
+          onClick={() => {
+            onFilter(Filter.ALL);
+          }}
```

Comment:
Do not use anonymous function.
Create separate handler that will cover all types

`const onClickHandler = (type: Filter) => onFilter(type);`

```suggestion
          onClick={onClickHandler(Filter.ALL)}
```

Reply:
that was an example
but you should create handlers for each case (be consistent)